src/lib/ui/promptService.js

```diff
@@ -0,0 +1,22 @@
+export default class PromptService {
+    constructor (promptProvider) {
+        this.promptProvider = promptProvider;
+    }
+
+    promptCartAbandonment() {
+        return new Promise((resolve, reject) => {
```

Comment:
Can this be written without wrapping it in another new Promise? Why not `await this.promptProvider.fire` or `return this.promptProvider.fire`?

Reply:
notice that `resolve` is called with `result.isConfirmed`

This is done so that none of the sweetalert API is exposed to the rest of the project. 
And that's just to make sure that if sweetalert change their API or we need to swap libraries for some reason, only this class has to change.